
THe Py_TRASHCAN_SAFE_BEGIN/END macros were deprecated in 3.11 and removed in
3.13.

https://github.com/python/cpython/issues/105111

diff -wpruN --no-dereference '--exclude=*.orig' a~/src/easy.c a/src/easy.c
--- a~/src/easy.c	1970-01-01 00:00:00
+++ a/src/easy.c	1970-01-01 00:00:00
@@ -280,13 +280,13 @@ PYCURL_INTERNAL void
 do_curl_dealloc(CurlObject *self)
 {
     PyObject_GC_UnTrack(self);
-    Py_TRASHCAN_SAFE_BEGIN(self);
+    Py_TRASHCAN_BEGIN(self, __func__);
 
     Py_CLEAR(self->dict);
     util_curl_close(self);
 
     Curl_Type.tp_free(self);
-    Py_TRASHCAN_SAFE_END(self);
+    Py_TRASHCAN_END;
 }
 
 
diff -wpruN --no-dereference '--exclude=*.orig' a~/src/multi.c a/src/multi.c
--- a~/src/multi.c	1970-01-01 00:00:00
+++ a/src/multi.c	1970-01-01 00:00:00
@@ -117,7 +117,7 @@ PYCURL_INTERNAL void
 do_multi_dealloc(CurlMultiObject *self)
 {
     PyObject_GC_UnTrack(self);
-    Py_TRASHCAN_SAFE_BEGIN(self);
+    Py_TRASHCAN_BEGIN(self, __func__);
 
     util_multi_xdecref(self);
     util_multi_close(self);
@@ -127,7 +127,7 @@ do_multi_dealloc(CurlMultiObject *self)
     }
 
     CurlMulti_Type.tp_free(self);
-    Py_TRASHCAN_SAFE_END(self);
+    Py_TRASHCAN_END;
 }
 
 
diff -wpruN --no-dereference '--exclude=*.orig' a~/src/share.c a/src/share.c
--- a~/src/share.c	1970-01-01 00:00:00
+++ a/src/share.c	1970-01-01 00:00:00
@@ -119,7 +119,7 @@ PYCURL_INTERNAL void
 do_share_dealloc(CurlShareObject *self)
 {
     PyObject_GC_UnTrack(self);
-    Py_TRASHCAN_SAFE_BEGIN(self);
+    Py_TRASHCAN_BEGIN(self, __func__);
 
     Py_CLEAR(self->dict);
     util_share_close(self);
@@ -133,7 +133,7 @@ do_share_dealloc(CurlShareObject *self)
     }
      
     CurlShare_Type.tp_free(self);
-    Py_TRASHCAN_SAFE_END(self);
+    Py_TRASHCAN_END;
 }
 
 
